### ISSUE TITLE
ConnectV4MappedIPAddressToV4Host_Success failed in CI on Linux

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#3494]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/2785/console
```
09:31:53    System.Net.Sockets.Tests.DualMode.ConnectV4MappedIPAddressToV4Host_Success [FAIL]
09:31:53       System.Net.Sockets.SocketException : Unknown error 10054
09:31:53       Stack Trace:
09:31:54             at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)
09:31:54             at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)
09:31:54             at System.Net.Sockets.Socket.Connect(IPAddress address, Int32 port)
09:31:54             at System.Net.Sockets.Tests.DualMode.DualModeConnect_IPAddressToHost_Helper(IPAddress connectTo, IPAddress listenOn, Boolean dualModeServer, Int32 port)
09:31:54             at System.Net.Sockets.Tests.DualMode.ConnectV4MappedIPAddressToV4Host_Success()
```

**Comment by stephentoub** [comment]
Also at http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/2785/

**Comment by stephentoub** [comment]
And the related ```ConnectV6IPEndPointToDualHost_Success``` failed at http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/2783/

**Comment by stephentoub** [comment]
Again here: http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/2810/

**Comment by stephentoub** [comment]
Again here: http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst/1128/console

I'm going to disable the test.

**Comment by ericeil** [comment]
Error 10054 is "connection reset."  What all these cases seem to have in common is that they involve accepting a connection, and then not cleanly closing the newly-accepted socket.  Perhaps Linux is a little less forgiving of this in the client-side `connect` call, resulting in an error where Windows reports success.

One solution might be to explicitly call `Shutdown` and `Dispose` on the accepted Socket.  Another would be to set the accepted socket to "linger" before discarding it.  Between those, I might favor the "linger" option, as it avoids a dependency on the more complex Shutdown code.

@pagavlin, does this sound right to you?  Any thoughts on the drawbacks of these solutions?

**Comment by ericeil** [comment]
Closed with #5907
